lumicks/pylake/__about__.py

```diff
@@ -1,5 +1,7 @@
+from importlib.metadata import version
+
 __title__ = "lumicks.pylake"
-__version__ = "1.5.1"
+__version__ = version("lumicks.pylake")
```

Comment:
I would swap this to work the other way around with `pyproject.toml` dynamic metadata: https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata

That way, `pyproject.toml` reads from this file rather than the other way around. IMO this is nicer because users will be doing `import lumicks.pylake` a lot more often than `pip install lumicks.pylake` so it's good to avoid extra overhead on the more frequently used path.

You can share `__summary__`, `__url__` and other info similarly.

Reply:
Yeah, I inverted the dependency since I didn't manage to make it work with the editable install (it complained about missing dependencies). But I only just realized that I should really just directly pull it from the `__about__` file instead. Must have been tired last Friday 😬.

This is actually nicer for a second reason on top of the overhead, since I would indeed want the package code to be leading in the case of an editable install, rather than the time I happened to install it.

I don't think `__summary__` and `__url__` can be shared through an attribute, and I would prefer not to put them in a separate file, since I feel that adding file reading to `__init__` or `__about__` is messier than having these properties we virtually never change in two places.